src/tests/test_connection.py

```diff
@@ -576,3 +577,216 @@ def test_feedback_when_switching_connection_without_alias(ip_empty, tmp_empty, c
 
     captured = capsys.readouterr()
     assert "Switching to connection duckdb://" == captured.out.replace("\n", "")
+
+
+@pytest.fixture
+def conn_sqlalchemy():
+    conn = SQLAlchemyConnection(engine=create_engine("duckdb://"))
+    yield conn
+    conn.close()
+
+
+@pytest.fixture
+def conn_dbapi():
+    conn = DBAPIConnection(duckdb.connect())
+    yield conn
+    conn.close()
+
+
+@pytest.fixture
+def mock_sqlalchemy_raw_execute(conn_sqlalchemy, monkeypatch):
+    mock = Mock()
+    monkeypatch.setattr(conn_sqlalchemy, "_connection_sqlalchemy", mock)
+    # mock the dialect to pretend we're using tsql
+    monkeypatch.setattr(conn_sqlalchemy, "_get_sqlglot_dialect", lambda: "tsql")
+
+    yield mock.execute, conn_sqlalchemy
+
+
+@pytest.fixture
+def mock_dbapi_raw_execute(monkeypatch, conn_dbapi):
+    mock = Mock()
+    monkeypatch.setattr(conn_dbapi, "_connection", mock)
+    # mock the dialect to pretend we're using tsql
+    monkeypatch.setattr(conn_dbapi, "_get_sqlglot_dialect", lambda: "tsql")
+
+    yield mock.cursor().execute, conn_dbapi
+
+
+@pytest.mark.parametrize(
+    "fixture_name",
+    [
+        "mock_sqlalchemy_raw_execute",
+        "mock_dbapi_raw_execute",
+    ],
+)
+def test_raw_execute_doesnt_transpile_sql_query(fixture_name, request):
+    mock_execute, conn = request.getfixturevalue(fixture_name)
+
+    conn.raw_execute("CREATE TABLE foo (bar INT)")
+    conn.raw_execute("INSERT INTO foo VALUES (42), (43)")
+    conn.raw_execute("SELECT * FROM foo LIMIT 1")
+
+    calls = [
+        str(call[0][0])
+        for call in mock_execute.call_args_list
+        # if running on sqlalchemy 1.x, the commit call is done via .execute,
+        # ignore them
+        if str(call[0][0]) != "commit"
+    ]
+
+    expected_number_of_calls = 3
+    expected_calls = [
+        "CREATE TABLE foo (bar INT)",
+        "INSERT INTO foo VALUES (42), (43)",
```

Comment:
is there a way to verify that original query is same as query being committed?

Reply:
that's what I'm doing here, I'm mocking the db connection objects and checking if the query that the user submitted is the same that the db executed